### ISSUE TITLE
[Yufei's dev computer environment breaks : compilation error using Maven]

### ISSUE THREAD

**Issue by imaffe** [imaffe/external-workspace-manager-plugin#1]
The external-workspace-manager-plugin on my computer cannot build locally, and my operations are : 
1. run `mvn test`
2. Print the following  error message 
> [ERROR]  Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.6.1:compile (default-compile) on project external-workspace-manager: Compilation failure
[ERROR] java.io.EOFException

but this is even stranger, as when 1 month ago it printed another message which I cannot reproduce now. I run  `mvn test` in another plugin i cloned , the aws-credentials-plugin, it generated the message as : 
> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project aws-credentials: Compilation failure
[ERROR] java.nio.file.NoSuchFileException: /home/affe/GitHubRes/aws-credentials-plugin/target/classes/META-INF/annotations/hudson.Extension

I asked in the chat and someone met this problem before. He upgraded the jenkins version and I did the same thing, but it doesn't work. Do you have any ideas on this ?

I now have some options to solve this : 
1. spend time on solving it, but cannot estimate the time to process it as I've already did a lot to try to fix this.
2. Reinstall my Ubuntu operating system and see what  happens.
3. Write code on dev computer, build on another computer.
4. Use my another computer as dev, but needs to configure environment and it is a little bit slow.

Currently I would like to try option 1 with your help. If 2 days before community bonding period, this still not solved, I'll try option 2 3 4 to see which one is better. Now I'm working on reading AWS SDK documentations, pure coding seems not that hard, but build and running will be time consuming.

**Comment by martinda** [comment]
Let's check the maven version. Mine says:
```
$ mvn --version
Apache Maven 3.3.9
Maven home: /usr/share/maven
Java version: 1.8.0_131, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-8-oracle/jre
Default locale: en_CA, platform encoding: UTF-8
OS name: "linux", version: "4.15.0-47-generic", arch: "amd64", family: "unix"
```
And `mvn test` works for me with master at 08073fc.

**Comment by imaffe** [comment]
I am going to change my Java version back to Java 8, and my Maven version is 3.6.0 (I remember I tried Java 8 but still not working, let's see what's gonna happen)

EDIT : I changed the Java version maven used to Java 8, and this fixed the second problem, which menas if I discovered this earlier, both of them would be fixed.

But now for the EOF exception, it still breaks. My git status says it is on 08073fc as well, so not likely to be I modified a file unconsciously. Thanks anyway we made some progress~


EDIT 2 : I deleted the repo and redownloaded it and thankfully, it worked out fine now. Thanks so much Martin~ Looks like its my bash Java version is inconsistent with JAVA_HOME version, which lead to this problem, and Maven use JAVA_HOME instead of bash Java source to decided its Java version. Thanks so much, I can write some code and build/test locally now ~ Wow that's COOL Martin ~
